### PR TITLE
raise AttributeError if predict_proba is not available

### PR DESCRIPTION
**Pull request by kmike**
In scikit-learn when predit_proba method is not available, AttributeError is raised instead of NotImplementedError. In this PR:
- classifiers are changed to follow the same convention;
- removed predict_log_proba mentions because lightning doesn't provide this method;
- added more tests for predict_proba results.

### CODE REVIEW

**Comment by kmike** [comment]
Python 2.7 and Python 3.5 tests are failing on Travis, but failures don't look related to the PR; Python 3.4 passes on Travis.

**Comment by fabianp** [comment]
I've relaunched Travis, lets see what happens

**Comment by kmike** [comment]
weird, it now fails with the same error as https://travis-ci.org/scikit-learn-contrib/lightning/jobs/171807463, but in a different environment

**Comment by fabianp** [comment]
OK I fixed a bug that I introduced in my recent pull request, lets see if now tests pass

**Comment by fabianp** [comment]
all tests pass now :-)

**Comment by fabianp** [comment]
This looks good to me. I'll give @mblondel some time to give his opinion. 

Thanks for your contribution!

File: lightning/impl/base.py [hunk 1]
```diff
@@ -52,7 +59,7 @@ def predict_proba(self, X):
             prob += 1
             prob /= 2
         else:
-            raise NotImplementedError("predict_(log_)proba only supported when"
+            raise NotImplementedError("predict_proba only supported when"
```
Comment (kmike) [inline lightning/impl/base.py]: I've kept NotImplementedError here as a defensive measure, but probably it is better to remove it.

**Comment by mblondel** [comment]
LGTM.

Could you also take a look at #18?

File: lightning/impl/base.py (hunk 1 above)
Comment (fabianp) [inline lightning/impl/base.py]: Why do you think it would be better to remove it?

Comment (kmike) [inline lightning/impl/base.py]: Because code flow should never get to this exception - AttributeError is raised earlier. If for some reason _predict_proba is used instead of predict_proba then with unsupported loss function an exception is still raised (`prob` variable is not defined); it is harder to interpret that NotImplementedError though. So maybe keeping it is fine :)

**Comment by fabianp** [comment]
2 LGTM = merge :-)

**Comment by kmike** [comment]
Thanks! 

@mblondel sorry; I may take a look at it at some point in future, but not right now.

**Comment by mblondel** [comment]
Sure, no worries. I was just trying to motivate someone to fix a long-standing issue :b

**Comment by fabianp** [comment]
@mblondel : are you OK if I make a quick release ASAP?

**Comment by mblondel** [comment]
@fabianp Sure, thanks for doing this!
